app/controllers/application_controller.rb

```diff
@@ -5,6 +5,7 @@ class ApplicationController < ActionController::Base
   before_action :set_current_user
   before_action :redirect_unauthenticated
   before_action :access_control
+  before_action :shibboleth_attributes
```

Comment:
Seems unnecessary to initialize a new user for *every* controller request. In particular, it'll always run, since it has no nil-handling.